soroban-env-host/src/native_contract/token/allowance.rs

```diff
@@ -5,11 +5,18 @@ use crate::native_contract::token::storage_types::{AllowanceDataKey, DataKey};
 use crate::{err, HostError};
 use soroban_env_common::{Env, StorageType, TryIntoVal};
 
+use super::storage_types::AllowanceValue;
+
 // Metering: covered by components
 pub fn read_allowance(e: &Host, from: Address, spender: Address) -> Result<i128, HostError> {
     let key = DataKey::Allowance(AllowanceDataKey { from, spender });
-    if let Ok(allowance) = e.get_contract_data(key.try_into_val(e)?, StorageType::Persistent) {
-        Ok(allowance.try_into_val(e)?)
+    if let Ok(allowance) = e.get_contract_data(key.try_into_val(e)?, StorageType::Temporary) {
+        let val: AllowanceValue = allowance.try_into_val(e)?;
+        if val.expiration_ledger <= e.get_ledger_sequence()?.into() {
```

Comment:
Should the check be only `<` instead of `<=` since the data is still usable on the expiration ledger, and only becomes unusable after it?

Related conversation: https://discord.com/channels/897514728459468821/1121913891996319795

Reply:
Updated based on the conversation we had.